### PR TITLE
Update subscriptions-core to 5.2.0

### PR DESCRIPTION
**Pull request by Jinksi**
#### Changes proposed in this Pull Request

Update `woocommerce/subscriptions-core` to `5.2.0`.
Most changes are related to adding HPOS support to subscriptions.

`composer.lock`'s change results from running `composer update woocommerce/subscriptions-core`.

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : **No changes to release testing/critical flows**
- [x] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.  **No changes to release testing/critical flows**
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

File: changelog/subscriptions-core-5.2.0-8 [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: patch
+Type: fix
+
+emoved the potential for an infinite loop when getting a subscription's related orders while the subscription is being loaded.
```
Comment (mattallan) [inline changelog/subscriptions-core-5.2.0-8]:
```suggestion
Removed the potential for an infinite loop when getting a subscription's related orders while the subscription is being loaded.
```

**Comment by Jinksi** [comment]
Note that I've conducted some basic WCPay subscription testing (with HPOS disabled / WC 7.3.0), with no issues identified.

**Review by mattallan — APPROVED** [review]
Smoke tests performed with WC Pay Subscriptions (WC Subscriptions deactivated):

**With HPOS both enabled and disabled:**
 - [x] Purchase multiple WC Pay Subscriptions
 - [x] Confirmed subscriptions in stripe dashboard
 - [x] View subscriptions on My Account
 - [x] Cancel subscription via My Account (confirmed changes in stripe dashboard)
 - [x] Resubscribe via My Account (confirmed new subscription in stripe dashboard)
 - [x] Change subscriptions payment method via My Account (confirmed new card in Stripe)
 - [x] View Subscriptions list table
 - [x] Suspend active Subscriptions using bulk actions on list table (confirmed subscriptions set with collection paused in Stripe)
 - [x] Activate subscription via Edit Subscription page (confirmed reactivated in Stripe)

**Comment by Jinksi** [comment]
I've tested the following flows to ensure these changes do not impact sites **with HPOS disabled**.

* WC 7.1.1 ([L-2 policy](https://developer.woocommerce.com/2021/05/12/woocommerce-payments-is-adopting-a-new-version-support-policy/))
* PHP 7.4

- [x] Create Variable Sub
- [x] Purchase Variable Sub alongside simple product
	- [x] Merchant and customer received new order email
- [x] Customer: My Account
	- [x] Order details are correct
	- [x] Related subscription link is displayed
	- [x] Subscription details are correct
	- [x] Related order link is displayed
- [x] Merchant
	- [x] Click link to sub in related orders metabox
	- [x] Sub is active. Details look correct
	- [x] Transaction in Payments → transactions looks correct
- [x] Enable Stripe test clocks on Sub (Dev testing mode)
- [x] Trigger `invoice.upcoming`  → `invoice.created`  → `invoice.paid`
	- [x] Renewal order appears **Processing**
	- [x] Merchant and customer received renewal order email
- [x] Trigger `invoice.upcoming`  → `invoice.created`  → `invoice.failed`
	- [x] Sub status changed to `on-hold` 
	- [x] Merchant and customer received renewal order failed email
- [x] Update payment method as customer (log in as Stripe Billing test customer)
	- [x] Payment method updated
	- [x] Successfully collected payment for your subscription using your new payment method.
	- [x] Sub status changed to `active` 
	- [x] Merchant and customer received renewal order success email 
- [x] Cancelled the subscription as merchant 
- [x] Resubscribe as customer via My Account 
	- [x] New resubscribe subscription created
	- [x] Resuscribe appears in the related orders metabox
	- [x] Initial subscription appears in the related orders metabox
- [x] Cancel subscription as customer via My Account
	- [x] Appears as “Pending Cancellation” to merchant 
- [x] View WC System Status report
	- [x] “Subscription Statuses” correct
	- [x] "Subscriptions by Payment Gateway” correct
- [x] Check related orders count on admin Subscriptions table

**Pull request merged**